### PR TITLE
add analog input readout

### PR DESCRIPTION
**Pull request by JennieYiChen**
## Add analog input readout
- Enables analog readout in two channels of mcusb
- Adds event callback handler that processes data from the mcusb readout buffer

### CODE REVIEW

File: analog_digital_out.py [hunk 1]
```diff
@@ -4,7 +4,9 @@
 """
 Sends ramping instructions to a MCUSB daq and live plots
 """
-import math, time, sys, os
+import math, time, sys, os, h5py, collections
+import numpy as np
+import matplotlib.pyplot as plt
```
Comment (dougUCN) [inline analog_digital_out.py]: Remove matplotlib plotting, out of the scope of this PR

File: analog_digital_out.py [hunk 2]
```diff
@@ -26,20 +38,48 @@ def main():
     mode = PUMP_MODE
     
     interface_type = InterfaceType.ANY
-    descriptor_index = 0
+    descriptor_index = 0 # Use the first detected device 
     
     # Parameters for AoDevice.a_out_scan
```
Comment (dougUCN) [inline analog_digital_out.py:42]: Change this comment to something clearer please

File: analog_digital_out.py [hunk 3]
```diff
@@ -26,20 +38,48 @@ def main():
...
-    #scan_options = ScanOption.SINGLEIO
-    #scan_flags = AOutScanFlag.DEFAULT
+    number_of_channels = ao_high_channel - ao_low_channel + 1
+    ao_samples_per_channel = 3000  
+    ao_sample_rate = 1000  # Hz
+    ao_scan_options = ScanOption.SINGLEIO
+    ao_scan_flags = AOutScanFlag.DEFAULT
     scan_status = ScanStatus.IDLE
     
+    # Parameters for AiDevice.a_in_scan
```
Comment (dougUCN) [inline analog_digital_out.py]: Change this comment to something more descriptive

File: analog_digital_out.py [hunk 4]
```diff
@@ -26,20 +38,48 @@ def main():
...
+    input_mode = AiInputMode.DIFFERENTIAL
+    range_index = 0
+    ai_samples_per_channel = 10000
+    ai_rate = 100
+    ai_available_sample_count = 2000
+    ai_scan_options = ScanOption.CONTINUOUS
+    ai_flags = AInScanFlag.DEFAULT
+    ai_channel_count = ai_high_channel - ai_low_channel + 1
+
+    # create a file to store data
```
Comment (dougUCN) [inline analog_digital_out.py]: Remove extraneous file creation comments

File: analog_digital_out.py [hunk 5]
```diff
@@ -53,26 +93,61 @@ def main():
...
+        # Get a list of supported ranges and validate the range index
+        ranges = ai_device.get_info().get_ranges(input_mode)
+        if range_index >= len(ranges):
+            range_index = len(ranges) - 1
+        
+        # Allocate a buffer to receive the data. Analog In
+        data = create_float_buffer(ai_channel_count, ai_samples_per_channel)
+       
+        # parameters in a_in_scan_events callback function
+        user_data = scan_params(
```
Comment (dougUCN) [inline analog_digital_out.py:113]: I think `callback_params` is a better name than `user_data`

File: analog_digital_out.py [hunk 6]
```diff
@@ -141,30 +223,127 @@ def main():
...
+    if (event_type == DaqEventType.ON_DATA_AVAILABLE
+            or event_type == DaqEventType.ON_END_OF_INPUT_SCAN):
+        
+        print('\n')
+        chan_count = user_data.ai_high_chan - user_data.ai_low_chan + 1
+        scan_count = event_data
+        total_samples = scan_count * chan_count
+
+        buffer_len = len(user_data.buffer) 
+        # clear_eol()
```
Comment (dougUCN) [inline analog_digital_out.py]: Remove extraneous comments

File: analog_digital_out.py (hunk 2 above)
Comment (JennieYiChen) [reply analog_digital_out.py:42]: It is probably better to let the user to choose which DAQ device they want? We might have more than one DAQ down the road.

File: analog_digital_out.py (hunk 1 above)
Comment (JennieYiChen) [reply analog_digital_out.py]: Got it! Thanks!

File: analog_digital_out.py (hunk 3 above)
Comment (JennieYiChen) [reply analog_digital_out.py]: Done!

File: analog_digital_out.py (hunk 2 above)
Comment (dougUCN) [reply analog_digital_out.py:42]: In my opinion no. That just becomes a complicated mess

File: analog_digital_out.py (hunk 4 above)
Comment (JennieYiChen) [reply analog_digital_out.py]: okok!

File: analog_digital_out.py (hunk 5 above)
Comment (JennieYiChen) [reply analog_digital_out.py:113]: I agree!

File: analog_digital_out.py (hunk 6 above)
Comment (JennieYiChen) [reply analog_digital_out.py]: done!

**Review by dougUCN — APPROVED** [review]
Go ahead and merge. For future reference try to be more descriptive on your commit messages

**Pull request merged**